### ISSUE TITLE
KubeCPUOvercommit doesn't take node pools

### ISSUE THREAD

**Issue by Duologic** [kubernetes-sigs/kubernetes-mixin#481]
https://github.com/kubernetes-monitoring/kubernetes-mixin/blob/dc563cbb03da396d23bc49f33d4f7ae28db514a4/alerts/resource_alerts.libsonnet#L25-L41

The KubeCPUOvercommit doesn't take node pools and tolerations into account and it might even be a stretch to cover that. Anyone has thoughts about that?

**Comment by brancz** [comment]
I have a feeling that only an effort like https://github.com/kubernetes/enhancements/pull/1916 could potentially go in this direction. Node pools are not well defined unfortunately, I actually poked around on sig-node yesterday if there might be possibilities to standardize this, but even if that were to lead to something, it's in the very beginning.

I think this would make sense for node-pools where you are aware of the scheduling constraints, I don't think going as far as tolerations is really reasonable or possible, as we would essentially implement the scheduler again I think.

**Comment by Duologic** [comment]
> I think this would make sense for node-pools where you are aware of the scheduling constraints, I don't think going as far as tolerations is really reasonable or possible, as we would essentially implement the scheduler again I think.

Agreed, probably sufficient to link a pod to a node pools through a recording rule.

**Comment by nhuray** [comment]
Hi guys,

Maybe my question is not directly related but I don't understand the expression of the alerting rule.

I have a very small cluster with just 1 instance and that rule is always firing because `(count(kube_node_status_allocatable_cpu_cores)-1)` returns 0. 

For example:

Let's say my container reserved 2 cpu on a single 4 cpu instance. The alerting rule give:

```
sum(namespace:kube_pod_container_resource_requests_cpu_cores:sum{%(ignoringOverprovisionedWorkloadSelector)s}) / 
               sum(kube_node_status_allocatable_cpu_cores)    = 0.5
                 > 
               (count(kube_node_status_allocatable_cpu_cores)-1) / count(kube_node_status_allocatable_cpu_cores)  = 0
```

but in reality I just commit 50% of the CPU resource of the entire cluster

**Comment by Duologic** [comment]
The alert message says:
>  ... and cannot tolerate node failure
So this alert is not applicable if you don't intend to run more than one node.

**Comment by nhuray** [comment]
I get your point but the message should be:

`Cluster has overcommitted CPU resource requests for Pods or cannot tolerate node failure.`
